core/src/apps/zcash/sig_hasher.py

```diff
@@ -0,0 +1,187 @@
+from trezor.crypto.hashlib import blake2b
+from trezor.utils import HashWriter, empty_bytearray
+from apps.bitcoin.writers import (
+    write_bytes_fixed,
+    write_bytes_reversed,
+    write_tx_output,
+    write_uint32,
+    write_uint64,
+    write_bytes_prefixed,
+    write_uint8,
+    TX_HASH_SIZE,
+)
+from apps.bitcoin import scripts
+from apps.bitcoin.common import SigHashType
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from trezor.messages import TxInput, TxOutput, SignTx, PrevTx
+    from trezor.utils import Writer
+    from apps.common.coininfo import CoinInfo
+    from typing import Sequence
+
+
+def write_hash(w: Writer, hash: bytes):
+    write_bytes_fixed(w, hash, TX_HASH_SIZE)
+
+
+def write_prevout(w: Writer, txi: TxInput):
+    write_bytes_reversed(w, txi.prev_hash, TX_HASH_SIZE)
+    write_uint32(w, txi.prev_index)
+
+
+class ZcashSigHasher:
```

Comment:
We have two classes named `ZcashSigHasher`, which is confusing, even though they are in different modules. The original one can be renamed to `Zip243SigHasher` or `ZcashLegacySigHasher`. The newer one can be renamed to `ZcashHasher` or `Zip244Hasher`, because it does more than just signature hashes (assuming we keep `txid_digest()`). There is also the question of whether we need it to be distinct from `ZcashExtendedSigHasher`.

In summary:
Change original `ZcashSigHasher` to `Zip243SigHasher` or `ZcashLegacySigHasher`.
Change new `ZcashSigHasher` to `Zip244Hasher` or `ZcashHasher` or merge with `ZcashExtendedSigHasher`.
Change `ZcashExtendedSigHasher` to `Zip244SigHasher` or `ZcashSigHasher`.

Reply:
I

- rename original `ZcashSigHasher` to `Zip243SigHasher`
- rename new `ZcashSigHasher` to `ZcashHasher`
- merge `ZcashExtendedSigHasher` into `ZcashHasher`

in https://github.com/trezor/trezor-firmware/pull/2166/commits/9cec3b7ed08d2c613717db28d6079759ea3c2712